src/System.Private.CoreLib/src/System/StartupHookProvider.cs

```diff
@@ -17,6 +14,22 @@ internal static class StartupHookProvider
         private const string StartupHookTypeName = "StartupHook";
         private const string InitializeMethodName = "Initialize";
 
+        private static readonly char[] s_disallowedSimpleAssemblyNameChars = new char[]
```

Comment:
No need to cache this for the duration of the process. It is used exactly once during startup.